signer/src/storage/model.rs

```diff
@@ -623,6 +623,30 @@ pub struct QualifiedRequestId {
     pub block_hash: StacksBlockHash,
 }
 
+/// This trait adds a function for converting a type into bytes to
+/// little-endian byte order. This is because stacks-core expects
+/// bitcoin block hashes to be in little-endian byte order when evaluating
+/// some clarity functions.
+///
+/// Both [`bitcoin::BlockHash`] and [`bitcoin::Txid`] are hash types that
+/// store bytes as SHA256 output, which is in big-endian order. Stacks-core
+/// stores hashes in little-endian byte order[2], implying that clarity
+/// functions, like `get-burn-block-info?`, return bitcoin block hashes in
+/// little-endian byte order. Note that Bitcoin-core transmits hashes in
+/// big-endian byte order[1] through the RPC interface, but the wire and
+/// zeromq interfaces transmit hashes in little-endian order[3].
+///
+/// [^1]: See the Note in
+///     <https://github.com/bitcoin/bitcoin/blob/62bd61de110b057cbfd6e31e4d0b727d93119c72/doc/zmq.md>.
+/// [^2]: <https://github.com/stacks-network/stacks-core/blob/70d24ea179840763c2335870d0965b31b37685d6/stacks-common/src/types/chainstate.rs#L427-L432>
+/// [^3]: <https://developer.bitcoin.org/reference/block_chain.html#block-chain>
+///       <https://developer.bitcoin.org/reference/p2p_networking.html>
+/// <https://learnmeabitcoin.com/technical/general/byte-order/>
+pub trait ToLittleEndianOrder: Sized {
```

Comment:
suggestion: same as above, would prefer it be more central for re-use.

Reply:
Yeah, we're due for a refactor. We can move this elsewhere